src/install/dracut-install.c

```diff
@@ -928,6 +967,19 @@ static int dracut_install(const char *orig_src, const char *orig_dst, bool isdir
                 } else {
                         log_info("cp '%s' '%s'", fullsrcpath, fulldstpath);
                         ret += cp(fullsrcpath, fulldstpath);
+
+                        if (arg_hash_source) {
+                                /* create hash of source and save it into the file */
+                                if (dst[0] != '/') {
+                                        if (_asprintf(&dstpath, "/%s", dst) < 0) {
```

Comment:
`_asprintf()` takes care of logging and exiting with failure in case of error. Drop the `if` block.